fastcrypto-zkp/src/api.rs

```diff
@@ -31,8 +33,13 @@ pub fn verify_groth16_in_bytes(
     proof_public_inputs_as_bytes: &[u8],
     proof_points_as_bytes: &[u8],
 ) -> Result<bool, FastCryptoError> {
-    let x = BlsFr::deserialize_compressed(proof_public_inputs_as_bytes)
-        .map_err(|_| FastCryptoError::InvalidInput)?;
+    let mut x = Vec::new();
+    for chunk in proof_public_inputs_as_bytes.chunks(SCALAR_LENGTH) {
+        if chunk.len() != SCALAR_LENGTH {
```

Comment:
you can check earlier if the `proof_public_inputs_as_bytes.len mod SCALAR_LENGTH` = 0 before parsing anything, to fail fast